ts/components/wallet/CanelPaymentLink.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * description
+ */
+
+import { Text } from "native-base";
+import * as React from "react";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+import { connect } from "react-redux";
+import I18n from "../../i18n";
+import { Dispatch } from "../../store/actions/types";
+import { paymentRequestCancel } from "../../store/actions/wallet/payment";
+
+type OwnProps = Readonly<{
+  navigation: NavigationScreenProp<NavigationState>;
+}>;
+
+type ReduxMappedDispatchProps = Readonly<{
+  cancelPayment: () => void;
+}>;
+
+type Props = OwnProps & ReduxMappedDispatchProps;
+
+class GoBackLink extends React.PureComponent<Props> {
+  public render(): React.ReactNode {
+    return (
+      <Text
+        bold={true}
+        white={true}
+        alignCenter={true}
+        onPress={() => this.props.cancelPayment()}
+      >
+        {I18n.t("wallet.ConfirmPayment.cancelPayment")}
+      </Text>
+    );
+  }
+}
+
+/**
+ *  goBack should have the same behaviour of the Cancel button of the screens related to the payment.
+ *  The effects of the story https://www.pivotaltracker.com/n/projects/2048617/stories/159427678
```

Comment:
is the reference to the story a TODO or it's supposed to be the description of this logic (in the latter case, better not write it here instead of referencing the story)

Reply:
It is a sort of TODO (hoping it can be resolved with the end of the review of this pr and the #447)